### ISSUE TITLE
False positive E251 about spaces around "=" for annotated parameters with default values in async functions 

### ISSUE THREAD

**Issue by east825** [PyCQA/pycodestyle#507]
According to PEP8 and #357 it's allowed to wrap with spaces equal sign before the default value of a parameter if the latter has annotation, e.g. `x: int = 42`. However if containing function is also marked as `async`, pep8 still falsely reports E251 "unexpected spaces around keyword / parameter equals" for its parameters.

For instance, for the next function pep8.py complains about whitespaces surrounding "=" sign. Meanwhile, if you remove `async` keyword, it now longer does.

```python
async def func(x: int = 42):
    pass
```

**Comment by sigmavirus24** [comment]
So https://github.com/PyCQA/pycodestyle/pull/361 fixed it (without the async keyword) *before* the async keyword was added. We can probably just update that fix to allow this to work properly.

**Comment by sigmavirus24** [comment]
I have something [here](https://github.com/PyCQA/pycodestyle/compare/master...bug/507) but GitHub isn't let me make a pull request bizarrely.

**Comment by sigmavirus24** [comment]
#508 should fix this.

**Comment by east825** [comment]
@sigmavirus24 Great! Thank you.
